jbpm-workitems/jbpm-workitems-rest/src/main/java/org/jbpm/process/workitem/rest/RESTWorkItemHandler.java

```diff
@@ -1343,6 +1366,29 @@ protected HttpClient getHttpClient(Integer readTimeout,
         }
     }
     
+    protected BasicCookieStore addCookie(String cookie,String url) {
+        String host=null;
+        BasicCookieStore basicCookieStore = new BasicCookieStore();
+        for (String c : cookie.split(";")) {
+            String[] cookieParts = c.split("=", 2);
+            if (cookieParts.length == 2) {
+                BasicClientCookie basicClientCookie = new BasicClientCookie(cookieParts[0], cookieParts[1]);
+                try {
+                    host=new URL(url).getHost();
+                } catch (MalformedURLException e) {
+                    host=null;
+                    logger.error("Exception occured while retriving host",e);
+                   
+                }
```

Comment:
Each cookie has doamin and path. If I move this block outside the for loop then I have to add another for loop to add domain path in case multiple cookies are passed. please share your thoughts

Reply:
Now it's fine, notice that only one domain per all cookies is supported

Reply:
ack